test/e2e/testnet/testnet.go

```diff
@@ -30,39 +30,25 @@ type Testnet struct {
 	knuu      *knuu.Knuu
 }
 
-func New(ctx context.Context, name string, seed int64, grafana *GrafanaInfo, chainID string, genesisModifiers ...genesis.Modifier) (*Testnet, error) {
-	identifier := fmt.Sprintf("%s_%s", name, time.Now().Format("20060102_150405"))
-	kn, err := knuu.New(ctx, knuu.Options{
-		Scope:        identifier,
-		ProxyEnabled: true,
-		// if the tests timeout, pass the timeout option
-		// Timeout: 120 * time.Minute,
-	})
-	if err != nil {
-		return nil, err
-	}
-
-	log.Info().
-		Str("scope", kn.Scope).
-		Str("TestName", name).
-		Msg("Knuu initialized")
-
-	kn.HandleStopSignal(ctx)
+type Options struct {
+	Seed             int64
+	Grafana          *GrafanaInfo
+	ChainID          string
+	GenesisModifiers []genesis.Modifier
+	Knuu             *knuu.Knuu
+}
 
+func New(opts Options) (*Testnet, error) {
```

Comment:
nit: I prefer the use of `Options` for arguments to a constructor which are in fact optional. All fields here can have working defaults except for the Knuu instance which is mandatory thus I'd prefer 

```suggestion
func New(knuu *knuu.Knuu, opts Options) (*Testnet, error) {
```

so now it's more explicit or better still if you go with

```go
type Option func (*Testnet) 

func WithSeed(seed int64) Option {
    return func (t *Testnet) {
        t.seed = seed
    }
}

func New(knuu *knuu.Knuu, options Option...) *Testnet
```

Reply:
yes the term "option" is kinda counter intuitive here. The main reason for using it to reduce the number of arguments to the constructor. 
Function option could be another way of using it; it is subjective, but I think option alone is cleaner than than the function option way.
we could extract `knuu` out of that option thing. a question might arise here: do we have default values for the rest of options if we extract knuu out? if so, I will do that.

Reply:
Here's a guess at defaults for the options:

```
	Seed             int64 => 42
	Grafana          *GrafanaInfo => &GrafanaInfo{}
	ChainID          string => test-chain-id
	GenesisModifiers []genesis.Modifier => []genesis.Modifier{}
```

Reply:
ok thanks Rootul, then let's do it

Reply:
done